test/NuGet.Core.Tests/NuGet.Commands.Test/SourceRepositoryDependencyProviderTests.cs

```diff
@@ -772,6 +774,141 @@ public async Task FindLibraryAsync_WhenASourceIsInaccessible_AndFailuresAreIgnor
             secondTestLogger.ShowWarnings().Should().Contain("NU1801");
         }
 
+        [Fact]
+        public async Task GetDependenciesAsync_WhenPackageHasVariousDependencyGroups_CorrectFrameworkIsSelected()
+        {
+            // Arrange
+            var testLogger = new TestLogger();
+            var cacheContext = new SourceCacheContext();
+            var findResource = new Mock<FindPackageByIdResource>();
+            var net46 = FrameworkConstants.CommonFrameworks.Net46;
+            var netstandard20 = FrameworkConstants.CommonFrameworks.NetStandard20;
+
+            var packageDependencyGroups = new List<PackageDependencyGroup>();
+            var net46group = new PackageDependencyGroup(net46, new PackageDependency[] { new PackageDependency("full.framework", VersionRange.Parse("1.0.0")) });
+            var netstandardGroup = new PackageDependencyGroup(netstandard20, new PackageDependency[] { new PackageDependency("netstandard", VersionRange.Parse("1.0.0")) });
+
+            packageDependencyGroups.Add(net46group);
+            packageDependencyGroups.Add(netstandardGroup);
+
+            findResource.Setup(s => s.GetDependencyInfoAsync(
+                    It.IsAny<string>(),
+                    It.IsAny<NuGetVersion>(),
+                    It.IsAny<SourceCacheContext>(),
+                    It.IsAny<ILogger>(),
+                    It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new FindPackageByIdDependencyInfo(
+                    new PackageIdentity("x", NuGetVersion.Parse("1.0.0-beta")),
+                    packageDependencyGroups,
+                    Enumerable.Empty<FrameworkSpecificGroup>()));
+
+            var source = new Mock<SourceRepository>();
+            source.Setup(s => s.GetResourceAsync<FindPackageByIdResource>())
+                .ReturnsAsync(findResource.Object);
+            source.SetupGet(s => s.PackageSource)
+                .Returns(new PackageSource("http://test/index.json"));
+
+            var libraryRange = new LibraryRange(
```

Comment:
Is this `libraryRange` used somewhere?

Reply:
Nope, remnant of a previous test version.
nice catch.